react-components/src/hooks/useSearchAssetsMapped360Annotations.tsx

```diff
@@ -0,0 +1,149 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import {
+  type AnnotationFilterProps,
+  type FileFilterProps,
+  type AnnotationsBoundingVolume,
+  type Asset,
+  type CogniteClient,
+  type AnnotationModel
+} from '@cognite/sdk';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+import { chunk, uniqBy } from 'lodash';
+
+export const useAllAssetsMapped360Annotations = (
+  sdk: CogniteClient,
+  siteIds: string[]
+): UseQueryResult<Asset[]> => {
+  return useQuery(
+    ['reveal', 'react-components', 'all-assets-mapped-360-annotations', siteIds],
+    async () => {
+      const assetMappings = await getAssetsMapped360Annotations(sdk, siteIds);
+      return assetMappings;
+    },
+    {
+      staleTime: Infinity
+    }
+  );
+};
+
+export const useSearchAssetsMapped360Annotations = (
+  siteIds: string[],
+  sdk: CogniteClient,
+  query: string
+): UseQueryResult<Asset[]> => {
+  const { data: assetMappings, isFetched } = useAllAssetsMapped360Annotations(sdk, siteIds);
+
+  return useQuery(
+    ['reveal', 'react-components', 'search-assets-mapped-360-annotations', query, siteIds],
+    async () => {
+      if (query === '') {
+        return assetMappings;
+      }
+
+      const filteredSearchedAssets =
+        assetMappings?.filter((asset) => {
+          const isInName = asset.name.toLowerCase().includes(query.toLowerCase());
+          const isInDescription = asset.description?.toLowerCase().includes(query.toLowerCase());
+
+          return isInName || isInDescription;
+        }) ?? [];
+
+      return filteredSearchedAssets;
+    },
+    {
+      staleTime: Infinity,
+      enabled: isFetched && assetMappings !== undefined
+    }
+  );
+};
+
+async function getAssetsMapped360Annotations(
+  sdk: CogniteClient,
+  siteIds: string[]
+): Promise<Asset[]> {
+  const fileIdsList = await get360ImagesFileIds(siteIds, sdk);
+  const image360Annotations = await get360ImageAnnotations(fileIdsList, sdk);
+  const result = await get360AnnotationAssets(image360Annotations, sdk);
+
+  return result;
+}
+
+async function get360AnnotationAssets(
+  image360Annotations: AnnotationModel[],
+  sdk: CogniteClient
+): Promise<Asset[]> {
+  const annotationMapping = image360Annotations.map((annotation) => ({
+    assetId:
+      (annotation.data as AnnotationsBoundingVolume).assetRef?.id ??
+      (annotation.data as AnnotationsBoundingVolume).assetRef?.externalId ??
+      undefined
+  }));
+
+  const uniqueAnnotationMapping = uniqBy(annotationMapping, 'assetId');
+
+  const assets = chunk(uniqueAnnotationMapping, 1000).map(async (uniqueAssetsChunk) => {
+    const retrievedAssets = await sdk.assets.retrieve(
+      uniqueAssetsChunk.map(({ assetId }) => {
+        if (typeof assetId === 'number') {
+          return { id: assetId };
+        } else if (typeof assetId === 'string') {
+          return { externalId: assetId };
+        } else {
+          throw new Error(`Invalid assetId: ${assetId}`);
```

Comment:
I think you should not throw here.  Because if you have a 360 image set where all except 1 annotation have assetRef set, then the whole set will be impossible to search through. I would suggest just initially filtering out all annotations that have undefined `assetRef`.